app/src/main/java/org/kiwix/kiwixmobile/webserver/ZimHostActivity.java

```diff
@@ -75,6 +82,7 @@
   private String ip;
   private ServiceConnection serviceConnection;
   private ProgressDialog progressDialog;
+  private boolean isActivityVisible;
```

Comment:
Not a fan of this variable at all, what is it achieving?

Reply:
@macgills so, when user clicks on the `Stop` button in the foreground notification. At that time, if activity is visible, we will update the layout Otherwise, we'll update it in `onResume()`.

Reply:
Just have it update. You can update views while your activity is in paused/stopped state

Reply:
But not when it is destroyed. Right?

Reply:
No but it is just going to go OnCreate>OnStart>OnResume when it is relaunched. If you are destroyed this code will not trigger anyway